src/SFA.DAS.EmployerIncentives.Commands/Persistence/AccountDomainRepository.cs

```diff
@@ -24,12 +26,26 @@ public async Task<Account> Find(long id)
         public Task Save(Account aggregate)
         {
             if (aggregate.IsNew)
-            {               
+            {
                 return _accountDataRepository.Add(aggregate.GetModel());
             }
 
             return _accountDataRepository.Update(aggregate.GetModel());
         }
+
+        public async Task<IEnumerable<Account>> GetByLegalEntityId(long legalEntityId)
```

Comment:
Please remove form the Domain repository. Use the existing query repository to find the account Id by other fields.  Then get the entity from the domain repository,  Treat them as separate read/write models with separate concerns.

Reply:
For what benefit?  It'll slow down the process and bloat the code.  Not being able to get a domain object by anything other than the primary key is a step too far imo.

Reply:
Because the domain repository is purely for hydrating and dehydrating domain objects form the logical write model.  Querying data for other purposes should be done through the existing query interface so that interface can be used elsewhere and indexed appropriately.  It keeps good separation of concerns and prevents the code from becoming a big mess.  It's a pattern following CQRS and although we don't have physical read/write models taking this approach removes the logical distinction.  However as this is supposedly a throw away project then if you want to add querying to this interface then I will resolve this.

Reply:
It's not querying per-se tho, it's merely retrieval of a domain object to update isn't it?